Documentation/security/threat-model.rst

```diff
@@ -0,0 +1,751 @@
+|                 |                     | network connectivity.        |
++-----------------+---------------------+------------------------------+
+
+.. _bandwidth limitations: https://docs.cilium.io/en/stable/network/kubernetes/bandwidth-manager/
+
+Recommended Controls
+^^^^^^^^^^^^^^^^^^^^
+
+-  Kubernetes workloads should have `defined resource limits`_.
+   This will help in ensuring that Cilium is not starved of resources due to a misbehaving deployment elsewhere in a cluster.
+-  Cilium can be given prioritized access to system resources either via
+   Kubernetes, cgroups, or other controls.
+-  Runtime security solutions such as `Tetragon`_ should be deployed to 
+   ensure that container compromises can be detected as they occur.
+
+.. _defined resource limits: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+.. _Tetragon: https://github.com/cilium/tetragon
+
+.. _limited-privilege-host-attacker:
+
+Limited-privilege Host Attacker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this scenario, the attacker is someone with the ability to run
+arbitrary code with direct access to the host PID or network namespace
+(or both), but without “root” privileges that would allow them to
+disable Cilium components or undermine the eBPF and other kernel state
+Cilium relies on.
+
+This level of access could exist for a variety of reasons, including:
+
+-  Pods or other containers running in the host PID or network
+   namespace, but not with “root” privileges. This includes
+   ``hostNetwork: true`` and ``hostPID: true`` containers.
+-  Non-“root” SSH or other console access to a Kubernetes worker node.
+-  A containerized workload that has “escaped” the container namespace
+   but as a non-privileged user.
+
+.. image:: images/cilium_threat_model_non_privileged.png
+
+In this case, an attacker would be able to bypass some of Cilium’s
+network controls, as described below:
+
+.. rst-class:: wrapped-table
+
++-----------------+-----------------------+----------------------------+
+| **Threat        | **Identified STRIDE   | **Cilium security          |
+| surface**       | threats**             | benefits**                 |
++=================+=======================+============================+
+| Cilium agent    | If the non-privileged |                            |
+|                 | attacker is able to   |                            |
+|                 | access the container  |                            |
+|                 | runtime and Cilium is |                            |
+|                 | running as a          |                            |
+|                 | container, the        |                            |
+|                 | attacker will be able |                            |
+|                 | to tamper with the    |                            |
+|                 | Cilium agent running  |                            |
+|                 | on the node.          |                            |
```

Comment:
I don't know if there's a better way to represent this threat, but this paragraph and the paragraphs under `Cilium configuration` and `Cilium eBPF programs` seem like the same threat to me. Do you consider them as different?

Possible suggestion, feel free to adopt or ignore at your discretion: The table makes it look like there are three threats. We might be able to use some sort of marker (`†`?) for the first instance and just write `Same as above (†)` in the rows below.